lib/vfscore/lookup.c

```diff
@@ -223,21 +227,37 @@ namei(const char *path, struct dentry **dpp)
 				node[0] = 0;
 
 				if (++links_followed >= MAXSYMLINKS) {
-					return (ELOOP);
+					error = ELOOP;
+					goto out;
 				}
 				need_continue = 1;
 				break;
 			}
 
 			if (*p == '/' && ddp->d_vnode->v_type != VDIR) {
 				drele(ddp);
-				return ENOTDIR;
+				error = ENOTDIR;
+				goto out;
 			}
 		}
 	} while (need_continue);
 
 	*dpp = dp;
-	return 0;
+out:
+	if (realpath)
+		strcpy(realpath, fp);
+	return error;
+}
+/*
+ * Convert a pathname into a pointer to a dentry
+ *
+ * @path: full path name.
+ * @dpp:  dentry to be returned.
+ */
+int
+namei(const char *path, struct dentry **dpp)
```

Comment:
```suggestion
int namei(const char *path, struct dentry **dpp)
```

Reply:
You can leave it on the same line, as mentioned in the coding - conventions PR: https://github.com/unikraft/docs/pull/248

> Prefer the Linux style with the return type on the same line as the function name.
In case of long functions + first argument, you can break after the return type.

Reply:
All of lookup.c uses this format, and I believe maintaining consistency is important.
If we however believe this style actively harms readability, we can address for the entire file in a style change PR.

In vfs.h: sure, there it makes sense, and the rest of the file follows suit as well.